airflow/models/baseoperator.py

```diff
@@ -393,27 +390,28 @@ def __init__(
                 .format(all_weight_rules=WeightRule.all_weight_rules,
                         d=dag.dag_id if dag else "", t=task_id, tr=weight_rule))
         self.weight_rule = weight_rule
+        self.resources: Optional[Resources] = None
+        if resources is not None:
+            # noinspection PyTypeChecker
```

Comment:
?

Reply:
Tried to solve ambigious type complaint by PyCharm. I will try to make it nicer.

Reply:
Improved.